src/background/stores/base-store-impl.ts

```diff
@@ -1,10 +1,14 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { HandlerReturnType } from 'common/flux/event-handler-list';
 import { BaseStore } from '../../common/base-store';
 import { Store } from '../../common/flux/store';
 import { StoreNames } from '../../common/stores/store-names';
 
-export abstract class BaseStoreImpl<TState> extends Store implements BaseStore<TState> {
+export abstract class BaseStoreImpl<TState, TReturn extends HandlerReturnType = void>
+    extends Store<TReturn>
+    implements BaseStore<TState>
```

Comment:
Could `TReturn` also be added to the BaseStore interface? I noticed that BaseStore's `addChangedListener` takes a `Function` with unspecified return type, and I wonder if that's why async store listeners are getting silently cast to () => void

Reply:
Yeah, we can definitely do that! It looks like it will be a lot of small changes though (We'll have to change `BaseStore<Type>` to `BaseStore<Type, HanlderReturnType>` in quite a lot of places) so I would prefer to do it as a separate PR